code/modules/holiday/aprilfools.dm

```diff
@@ -0,0 +1,279 @@
+	H.remove_overlay(HAIR_LAYER)
+	if(H.disabilities & HUSK)
+		return
+
+	var/list/standing = list()
+
+	if(pony_hair && (HAIR in specflags))
+		var/image/img_hair_s = image("icon" = 'code/modules/holiday/aprilfools.dmi', "icon_state" = "[pony_hair]_hair", "layer" = -HAIR_LAYER)
+		img_hair_s.color = "#" + H.hair_color
+		standing += img_hair_s
+
+	if(pony_tail && (HAIR in specflags))
+		var/image/img_hair_s = image("icon" = 'code/modules/holiday/aprilfools.dmi', "icon_state" = "[pony_hair]_tail", "layer" = -HAIR_LAYER)
+		img_hair_s.color = "#" + H.hair_color
+		standing += img_hair_s
+
+	if(standing.len)
+		H.overlays_standing[HAIR_LAYER]	= standing
+
+	H.apply_overlay(HAIR_LAYER)
+
+/datum/species/pony/handle_body(mob/living/carbon/human/H)
+	H.remove_overlay(BODY_LAYER)
+	var/list/standing	= list()
+
+	handle_mutant_bodyparts(H)
+	standing += image("icon" = 'code/modules/holiday/aprilfools.dmi', "icon_state" = "eyes", "layer" = -BODY_LAYER)
+
+	// eyes
+	if(EYECOLOR in specflags)
+		var/image/img_eyes_s = image("icon" = 'code/modules/holiday/aprilfools.dmi', "icon_state" = "eyes_inner", "layer" = -BODY_LAYER)
+		img_eyes_s.color = "#" + H.eye_color
+		standing	+= img_eyes_s
+
+	if(standing.len)
+		H.overlays_standing[BODY_LAYER] = standing
+
+	H.apply_overlay(BODY_LAYER)
+	return
+
+/datum/species/pony/handle_mutant_bodyparts(mob/living/carbon/human/H, forced_colour)
+	var/list/bodyparts_to_add = mutant_bodyparts.Copy()
+	var/list/relevent_layers = list(BODY_BEHIND_LAYER, BODY_ADJ_LAYER, BODY_FRONT_LAYER)
+	var/list/standing	= list()
+
+	H.remove_overlay(BODY_BEHIND_LAYER)
+	H.remove_overlay(BODY_ADJ_LAYER)
+	H.remove_overlay(BODY_FRONT_LAYER)
+
+	if(!mutant_bodyparts)
+		return
+
+	if(!bodyparts_to_add)
+		return
+
+	var/image/I
+
+	for(var/layer in relevent_layers)
+		for(var/bodypart in bodyparts_to_add)
+			var/icon_string = H.dna.features[bodypart]
+			if(!icon_string)
+				icon_string = default_features[bodypart]
+
+			if(!icon_string || icon_string == "None")
+				continue
+
+			I = image("icon" = 'code/modules/holiday/aprilfools.dmi', "icon_state" = icon_string, "layer" =- layer)
+
+			if(!(H.disabilities & HUSK))
+				switch(bodypart)
+					if("wings")
+						I.color = "#[H.dna.features["mcolor"]]"
+					if("hair", "tail")
+						I.color = "#[H.hair_color]"
+					if("alicorn")
+						if(alicorn_color)
+							I.color = alicorn_color
+			standing += I
+
+		H.overlays_standing[layer] = standing.Copy()
+		standing = list()
+
+	H.apply_overlay(BODY_BEHIND_LAYER)
+	H.apply_overlay(BODY_ADJ_LAYER)
+	H.apply_overlay(BODY_FRONT_LAYER)
+
+/datum/species/pony/spec_life(mob/living/carbon/human/H)
+	if(H.reagents.get_reagent_amount("friendship") < 100)
+		H.reagents.add_reagent("friendship", 2)
+	return
+
+/datum/species/pony/handle_speech(message, mob/living/carbon/human/H)
+	return message
+
+//return a list of spans or an empty list
+/datum/species/pony/get_spans()
+	return list()
+
+
+// FRIENDSHIP REAGENT
+// Sto- Ported from Goonstation.
```

Comment:
shipping datums are original tho

although yeah stocks already got in (with permission from marq, who coded it)